src/Components/Login.tsx

```diff
@@ -1,13 +1,39 @@
 import React, { useContext, useState } from 'react';
 import { UserContext } from '../Contexts/UserContext';
+import { registerUser } from '../network';
 
 export default function Login() {
+	const { setUsername, setHome } = useContext(UserContext);
 	const [formUsername, setFormUsername] = useState<string>('');
 	const [latitude, setLatitude] = useState<string>('');
 	const [longitude, setLongitude] = useState<string>('');
+	const [errorMessage, setErrorMessage] = useState<string>('');
+
+	function isValidLat(input: string): boolean {
+		const value = parseFloat(input);
+		return !isNaN(value) && value >= -90 && value <= 90;
+	}
+
+	function isValidLong(input: string): boolean {
+		const value = parseFloat(input);
+		return !isNaN(value) && value >= -180 && value <= 180;
+	}
 
 	function handleSubmit(e: React.FormEvent) {
 		e.preventDefault();
+
+		return registerUser(formUsername).then(
+			(newUser: null | { username: string }) => {
+				if (!newUser) {
+					setErrorMessage(
+						'Sorry, there was an error creating an account. Please try again later.'
+					);
+				} else {
+					setUsername(newUser.username);
+					setHome([parseFloat(latitude), parseFloat(longitude)]);
```

Comment:
Nothing happens after this - I wanted to go to the map , Steve! :(

Reply:
I assume that's because of the wrong endpoint url?

Reply:
Nope - It 204's now, content created, and doesnt redirect.